x/crs/keeper/keeper.go

```diff
@@ -64,3 +66,50 @@ func NewKeeper(cdc codec.BinaryCodec, addressCodec address.Codec, storeService s
 func (k Keeper) GetAuthority() string {
 	return k.authority
 }
+
+func (k Keeper) EndBlocker(ctx context.Context, id uint64) { // Added id as a parameter
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
```

Comment:
here we should first get all the decisions for which its reveal time has passed, and then go through the reveals of those decisions.